src/main/java/app/ultratechies/api/AppUsers/UserAuth/UserNameAuthController.java

```diff
@@ -0,0 +1,29 @@
+package app.ultratechies.api.AppUsers.UserAuth;
+
+
+import app.ultratechies.api.AppUsers.AppUser;
+import app.ultratechies.api.AppUsers.UserDTO.UserDto;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Optional;
+
+@RestController
+@RequestMapping("api/user/name/auth")
+public class UserNameAuthController {
+
+    private final UserNameAuthService userNameAuthService;
+
+    @Autowired
+    public UserNameAuthController(UserNameAuthService userAuthService) {
+        this.userNameAuthService = userAuthService;
+    }
+
+    @PostMapping()
+    public ResponseEntity<Optional<UserDto>> getUser(@RequestBody AppUser appUser){
```

Comment:
@iannjari we should use the UserDto as the request body instead

Reply:
UserDto class does not have a password field, that's why